service/helmmigration/migration.go

```diff
@@ -0,0 +1,201 @@
+package helmmigration
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	chartName   = "default-catalog/helm-2to3-migration"
+	releaseName = "draughtsman-helm-migration"
+)
+
+type Config struct {
+	KubernetesClient kubernetes.Interface
+	Logger           micrologger.Logger
+
+	HelmBinaryPath string
+	ProjectList    []string
+	Repository     string
+}
+
+type HelmMigration struct {
+	kubernetesClient kubernetes.Interface
+	logger           micrologger.Logger
+
+	helmBinary  string
+	repository  string
+	projectList []string
+}
+
+func New(c Config) (*HelmMigration, error) {
+	if c.KubernetesClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "kubernetesClient must not be empty")
+	}
+	if c.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "logger must not be empty")
+	}
+
+	if c.HelmBinaryPath == "" {
+		return nil, microerror.Maskf(invalidConfigError, "helmBinaryPath must not be empty")
+	}
+	if c.Repository == "" {
+		return nil, microerror.Maskf(invalidConfigError, "repository must not be empty")
+	}
+	if c.ProjectList == nil {
+		return nil, microerror.Maskf(invalidConfigError, "projectList must not be empty")
+	}
+
+	h := HelmMigration{
+		kubernetesClient: c.KubernetesClient,
+		logger:           c.Logger,
+
+		repository:  c.Repository,
+		helmBinary:  c.HelmBinaryPath,
+		projectList: c.ProjectList,
+	}
+
+	return &h, nil
+}
+
+func (h *HelmMigration) Migrate(ctx context.Context) error {
+	h.logger.Debugf(ctx, "Checking remaining helm2 releases from draughtsman")
+	projectList, err := h.listRemainingHelmRelease()
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if len(projectList) == 0 {
+		h.logger.Debugf(ctx, "no helm2 releases from draughtsman, quitting now...")
+		return nil
+	}
+
+	h.logger.Debugf(ctx, "migrating total %d helm2 releases from draughtsman", len(projectList))
+
+	err = h.installHelm2to3Migration(projectList)
```

Comment:
Does it work when we have 2 charts installed at the same time?

Reply:
@kopiczko Do you mean we have 2 charts such as aws-operator, aws-app-collection? Then we don't have a problem since `helm-2to3-migration` app would migrate all apps one by one.

Reply:
No I thought about 2 charts of the same app. E.g aws-operator installed in both helm2 and helm3 at the same time. Does that work by adopting existing manifests? I guess you already exercised that during app platform migration so I trust you know what you're doing.

Reply:
We are using `2to3` utility tool which migrates all the helm2 resources into helm3. 
After migration, we deactivated helm2 releases by changing the labels in helm2 configmap. 
https://github.com/giantswarm/helm-2to3-migration/blob/7ae6fea0a4bf75b0b3220532001cf396332ed717/helm/helm-2to3-migration/templates/cm.yaml#L43
So we are not using helm2 and helm3 releases at the same time actually.